### PR TITLE
Format as package

### PR DESCRIPTION
**Pull request by kaitejohnson**
Not doing anything here but setting up infrastructure for the creation of an R package called `wwinference`. The goal will be to in the next week, port of just the modeling components from `cfaforecastrenewalww` and the immediate pre-processing from `wweval`. This PR should just provide the skeleton components for that package.

Next steps: 
- set up CI for testing suite

### CODE REVIEW

File: .github/.gitignore
Comment (dylanhmorris) [inline .github/.gitignore:1]: why is this needed?

File: .github/actions/pre-commit/action.yaml
Comment (dylanhmorris) [inline .github/actions/pre-commit/action.yaml:1]: Now that we're on CDCGov, we can and should use the official pre-commit action 

https://github.com/pre-commit/action

(Can't use it on `cdcent` since it's not on the auto-approved list)

Comment (dylanhmorris) [reply .github/actions/pre-commit/action.yaml:1]: This can be in this PR or in a separate one, as you prefer @kaitejohnson

Comment (dylanhmorris) [reply .github/actions/pre-commit/action.yaml:1]: For an example, see the multisignal project https://github.com/CDCgov/multisignal-epi-inference/blob/main/.github/workflows/pre-commit.yaml

File: .github/workflows/pkgdown.yaml
Comment (dylanhmorris) [inline .github/workflows/pkgdown.yaml:1]: Suggest separating into `build` and `deploy`, as we do here:

https://github.com/CDCgov/wastewater-informed-covid-forecasting/blob/prod/.github/workflows/pkgdown.yaml

File: .github/workflows/pre-commit.yaml
Comment (dylanhmorris) [inline .github/workflows/pre-commit.yaml:1]: See above comment on the custom action.

**Review by dylanhmorris — COMMENTED** [review]
Looks good. A few questions/comments @kaitejohnson

File: .github/.gitignore
Comment (kaitejohnson) [reply .github/.gitignore:1]: Good question, I think this was maybe generated when I tried to set up the `pkgdown` and I just left it in because I thought maybe there was a reason for it... we have something similar in our other repo... https://github.com/CDCgov/wastewater-informed-covid-forecasting/blob/prod/.github/.gitignore can delete though!

File: .github/actions/pre-commit/action.yaml
Comment (kaitejohnson) [reply .github/actions/pre-commit/action.yaml:1]: I can replace with this in this PR

File: .github/workflows/pkgdown.yaml
Comment (kaitejohnson) [reply .github/workflows/pkgdown.yaml:1]: I am guessing I shouldn't generate this by hand. Did you generate this using `usethis::use_github_pages()` or something else?

File: .github/workflows/pre-commit.yaml
Comment (kaitejohnson) [reply .github/workflows/pre-commit.yaml:1]: Getting an issue with styler... tried listing it as a dependency in the package but that didn't seem to work...

Comment (dylanhmorris) [reply .github/workflows/pre-commit.yaml:1]: Just needed to clear the cache and re-run. https://github.com/CDCgov/ww-inference-model/actions/runs/9705508350

Can remove it as a dependency.

Comment (kaitejohnson) [reply .github/workflows/pre-commit.yaml:1]: Hmmm I tried rerunning, am I missing something else that you need to do to clear the cache on github?

Comment (dylanhmorris) [reply .github/workflows/pre-commit.yaml:1]: Sorry, should've clarified. The caches here: https://github.com/CDCgov/ww-inference-model/actions/caches

File: .github/workflows/pkgdown.yaml
Comment (dylanhmorris) [reply .github/workflows/pkgdown.yaml:1]: `usethis::use_github_pages()` is nice for generating a starting workflow, but it's fine to hand-modify it.

File: .github/workflows/pre-commit.yaml
Comment (dylanhmorris) [reply .github/workflows/pre-commit.yaml:1]: Worked for me just now: https://github.com/CDCgov/ww-inference-model/actions/runs/9713107113

Comment (dylanhmorris) [reply .github/workflows/pre-commit.yaml:1]: Interesting, it fails when trying to use the cache. Are you sure the `.pre-commit config.yaml` is up to date with the latest version of the R hooks?

Comment (kaitejohnson) [reply .github/workflows/pre-commit.yaml:1]: I used the one on the template repo, let me compare it to some others...

Comment (dylanhmorris) [reply .github/workflows/pre-commit.yaml:1]: I just checked, it isn't that. I think I may know what it is. I'll branch from this branch and try something.

Comment (kaitejohnson) [reply .github/workflows/pre-commit.yaml:1]: Hmm okay. Perhaps we should make a cdcgov template repo that sets things up using the official pre-commit action?

Comment (kaitejohnson) [reply .github/workflows/pre-commit.yaml:1]: @dylanhmorris Adding in the args that I copied from https://github.com/CDCgov/cfa-gam-rt/blob/main/.pre-commit-config.yaml and rerunning locally seems to have solved it. Would seem you need the `args`?

Comment (dylanhmorris) [reply .github/workflows/pre-commit.yaml:1]: That succeeded b/c that invalidated the cache, but [re-running still fails](https://github.com/CDCgov/ww-inference-model/actions/runs/9713752653) because it tries to use the cache.

File: .github/workflows/pkgdown.yaml
Comment (kaitejohnson) [reply .github/workflows/pkgdown.yaml:1]: Ok I tried to modify to match the build and deploy set up...

File: .github/workflows/pre-commit.yaml
Comment (dylanhmorris) [reply .github/workflows/pre-commit.yaml:1]: Explicitly using [`actions/cache`](https://github.com/actions/cache) to retain the `renv` cache appears to [resolve the problem (note success on multiple successful re-runs)](https://github.com/CDCgov/ww-inference-model/actions/runs/9717464679)

File: .github/workflows/pkgdown.yaml [hunk 1]
```diff
@@ -38,38 +38,24 @@ jobs:
...
-
-      environment:
-        name: github-pages
-        url: ${{ steps.deployment.outputs.page_url }}
-
-      steps:
-        - name: Deploy to GitHub pages
-          uses: actions/deploy-pages@v2.0.4
+          clean: false
+          branch: gh-pages
```
Comment (dylanhmorris) [inline .github/workflows/pkgdown.yaml:60]: I would use environment deployment, not deployment to a github pages branch. [Relevant Github documentation on the distinction](https://docs.github.com/en/pages/getting-started-with-github-pages/configuring-a-publishing-source-for-your-github-pages-site#creating-a-custom-github-actions-workflow-to-publish-your-site).

Here's an example of an action for environment deployment: https://github.com/CDCgov/wastewater-informed-covid-forecasting/blob/prod/.github/workflows/pkgdown.yaml

**Comment by kaitejohnson** [comment]
Ok @dylanhmorris I think the issue I was having was with trying to install `cmdstanr`, while `cmdstar` is not a dependency (yet, it will be in my next PR). 

But would you take a look at the current version of `pkgdown.yml` and see if this is a good starting place? I modified it from https://github.com/CDCgov/wastewater-informed-covid-forecasting/blob/prod/.github/workflows/pkgdown.yaml

File: .github/workflows/pkgdown.yaml (hunk 1 above)
Comment (kaitejohnson) [reply .github/workflows/pkgdown.yaml]: Ok repeating back in other words, we want to trigger a change to the page whenever a PR is merged into main, we don't want to make a new branch called `gh-pages`  and say have to merge to that to get updates

File: .github/.gitignore
Comment (kaitejohnson) [reply .github/.gitignore:1]: Removed

File: .github/workflows/pkgdown.yaml
Comment (kaitejohnson) [reply .github/workflows/pkgdown.yaml:1]: Let me know what you think of the current set up

File: .github/workflows/pkgdown.yaml (hunk 1 above)
Comment (dylanhmorris) [reply .github/workflows/pkgdown.yaml]: First sentence is correct. RE: second: an older Github Pages workflow was to store rendered website files in a dedicated branch  in the repo (typically named `gh-pages`). That has now been superseded by an approach where they're stored outside the repo itself. One advantage of this second approach is that it more clearly separates the pre-rendered source (stored in the repo, version-controlled) from the rendered website files (stored outside it, not necessarily version-controlled)

File: .github/workflows/pkgdown.yaml [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+      - uses: r-lib/actions/setup-r-dependencies@v2
+        with:
+          pak-version: rc
+          extra-packages: any::pkgdown local::.
+          needs: website
+
+      - name: Build site
+        run: "pkgdown::build_site_github_pages(new_process = FALSE, install = FALSE)"
+        shell: Rscript {0}
+
```
Comment (dylanhmorris) [inline .github/workflows/pkgdown.yaml:48]: Need to upload the artifact at the end of this. Compare 

https://github.com/CDCgov/wastewater-informed-covid-forecasting/blob/ce70b29453f46cac44b1100f6f46d11624db2715/.github/workflows/pkgdown.yaml#L55-L58

**Review by dylanhmorris — COMMENTED** [review]
A couple comments on the `pkgdown` workflow @kaitejohnson

File: .github/workflows/pkgdown.yaml (hunk 2 above)
Comment (kaitejohnson) [reply .github/workflows/pkgdown.yaml:48]: Ok I think I am missing something here, but I am getting the error:
```
tar: wwinference/docs/: Cannot open: No such file or directory
tar: Error is not recoverable: exiting now
```

In the example from the other ww repo, we don't have `docs` folder within the repo either? So I am not sure what is missing here, I guess I thought this would somehow be generated with what is called above...

Comment (dylanhmorris) [reply .github/workflows/pkgdown.yaml:48]: Yes, it will be generated dynamically. Here it will just be called `docs/` because there isn't a subdirectory.

Comment (dylanhmorris) [reply .github/workflows/pkgdown.yaml:48]: You can try running 

```bash
Rscript "pkgdown::build_site_github_pages(new_process = FALSE, install = FALSE)"
```

at your local command line from the top-level project directory. A `docs/` folder should be generated.

**Comment by kaitejohnson** [comment]
Ok I think this is now ready

**Comment by dylanhmorris** [comment]
Apologies for the delay. Reviewing now.

**Review by dylanhmorris — APPROVED** [review]
LGTM
